implementations/rust/channel/src/lib.rs

```diff
@@ -247,6 +247,11 @@ impl<I: KeyExchanger, R: KeyExchanger, E: NewKeyExchanger<I, R>> ChannelManager<
 
         let channel = self.channels.get_mut(&cipher_address).unwrap();
         let mut channel = &mut *channel.lock().unwrap();
+        let agreement = match &mut channel.agreement {
+            Some(e) => e.as_mut(),
+            None => panic!(), // FIXME
```

Comment:
We shouldn't panic. Return an error instead.

Reply:
Agree, this will be fixed in actual mergeable PRs